src/libraries/Common/src/System/Text/Base64UrlEncoding.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text
+{
+    internal static class Base64UrlEncoding
```

Comment:
We've had multiple requests for this functionality, e.g. https://github.com/dotnet/runtime/issues/1658.  How close is this to what's needed for that?  I'm wondering if we should use this as an opportunity to get it into the right shape and move that issue to api-ready-for-review so we can get it out in 7.